packages/loot-core/src/server/budget/goaltemplates.js

```diff
@@ -329,19 +329,25 @@ async function applyCategoryTemplate(category, template_lines, month, force) {
         break;
       }
       case 'percentage': {
-        /*
-          let income_category = (await actual.getCategories()).filter(c => c.is_income == true && c.name == template.category);
-          let func = (getBudgetMonthTestFunc || getBudgetMonth);
-          let budget = await func(month);
-          for (var g = 0; g < budget.categoryGroups.length; g++) {
-            if (income_category.group_id == budget.categoryGroups[g].id) {
-              for (var c = 0; c < budget.categoryGroups[g].categories.length; c++)
-                if (income_category.id == budget.categoryGroups[g].categories[c].id) {
-                  let month_category = budget.categoryGroups[g].categories[c];
-                }
-            }
+        let percent = template.percent;
+        let monthlyIncome = 0;
+        if (template.category === 'all income') {
+          monthlyIncome = await getSheetValue(sheetName, `total-income`);
+        } else {
+          let income_category = (await db.getCategories()).filter(
+            c => c.is_income == true && c.name == template.category,
+          );
+          if (!income_category) {
+            throw new Error(`Could not find category “${template.category}”`);
```

Comment:
Also I know I suggested adding this but we currently don’t handle errors well here. We should update the `applyCategoryTemplate` API to return a value or an error, and then update all the places where `console.log` is currently called to return an error too. This should probably be done in a different PR that will merge before this one, though.

Reply:
⏩ 859

Reply:
Can you update this to work with the new error logic?

Reply:
I will.  I merged the master in already, so I'm just waiting to get back to a PC.